src/Components/Common/MarkdownEditor.tsx

```diff
@@ -0,0 +1,396 @@
+import { useRef, useState, useEffect } from "react";
+import {
+  FaBold,
+  FaItalic,
+  FaListOl,
+  FaListUl,
+  FaUndo,
+  FaRedo,
+  FaLink,
+  FaUnlink,
+  FaQuoteRight,
+} from "react-icons/fa";
+import TurndownService from "turndown";
+import ReactMarkdown from "react-markdown";
+
+interface RichTextEditorProps {
+  markdown?: string;
+  onChange?: (markdown: string, htmlCode: string) => void;
+}
+
+const RichTextEditor: React.FC<RichTextEditorProps> = () => {
+  const [markdown, setMarkdown] = useState<string>("");
+  const [htmlCode, setHtmlCode] = useState<string>("");
+  const editorRef = useRef<HTMLDivElement>(null);
+  const undoStack = useRef<string[]>([]);
+  const redoStack = useRef<string[]>([]);
+  const [isBoldActive, setIsBoldActive] = useState<boolean>(false);
+  const [isItalicActive, setIsItalicActive] = useState<boolean>(false);
+  const [isQuoteActive, setIsQuoteActive] = useState<boolean>(false);
+  const [isUnorderedListActive, setIsUnorderedListActive] =
+    useState<boolean>(false);
+  const [isOrderedListActive, setIsOrderedListActive] =
+    useState<boolean>(false);
+
+  useEffect(() => {
+    document.addEventListener("selectionchange", handleSelectionChange);
+    return () => {
+      document.removeEventListener("selectionchange", handleSelectionChange);
+    };
+  }, []);
+
+  const handleSelectionChange = () => {
+    const selection = window.getSelection();
+    if (!selection || !selection.rangeCount) return;
+
+    setIsBoldActive(document.queryCommandState("bold"));
+    setIsItalicActive(document.queryCommandState("italic"));
+    setIsQuoteActive(
+      selection.focusNode?.parentElement?.tagName === "BLOCKQUOTE",
+    );
```

Comment:
There seem to be minor bugs while enabling styles (like bold, ...) without selecting the text.